java/spark/src/main/java/com/lancedb/lance/spark/LanceIdentifier.java

```diff
@@ -14,13 +14,85 @@
 package com.lancedb.lance.spark;
 
 import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
 
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * LanceIdentifier is a custom implementation of {@link Identifier} for Lance. It contains the
+ * dataset URI and the namespace. The namespace is an array of strings, which contains the namespace
+ * of the dataset and the options. The options are key-value pairs, which are separated by "#####".
```

Comment:
What is the purpose of the namespace? Why does it store a serialized copy of the options?

Reply:
The namespace was stand for catalog and database name. In this case, the catalog load table and create table will convert Lance Identifier into IdentifierImpl which only have namespaces and name. The way Iceberg does is add options in the path stored in name. I store the options in namespaces.

Reply:
Maybe we can wait until the design of Lance Catalog landing?

Reply:
The datasource api is not depend on catalog. 
The AI user maybe do not use the catalog.

Reply:
Ok, I checked this file has existed before, not introduced recently. Will review it later.